lib/kits/core-ui/components/common/offer-list.js

```diff
@@ -155,15 +166,36 @@ const OfferList = ({publicationType, configs = {}, sgnViewer, template}) => {
                     'click',
                     publicationType === 'incito'
                         ? scrollToOffer
-                        : redirectToPage,
+                        : navigateToPage,
                     false
                 );
             });
     };
 
-    const redirectToPage = (e) => {
+    const navigateToPage = (e) => {
+        e.preventDefault();
+        const {pageId, pageNum} = e.currentTarget.dataset;
+
+        if (scriptEls.displayUrlParams?.toLowerCase() === 'query') {
+            pushQueryParam(
+                [
+                    {
+                        name: scriptEls.publicationIdParam,
+                        val: configs.id
+                    },
+                    {
+                        name: scriptEls.pageIdParam,
+                        val: pageNum
+                    }
+                ],
+                pageNum ? 'set' : 'delete'
+            );
```

Comment:
I think this `pushQueryParam` API is a bit clunky, how do you feel about a call signature that would make this part look like this instead:
```js
pushQueryParam({
  [scriptEls.publicationIdParam]: configs.id,
  [scriptEls.pageIdParam]: pageNum || undefined,
});
```

(`undefined` then means delete the param)